### PR TITLE
KAFKA-7904: Add AtMinIsr partition metric and TopicCommand option (KIP-427)

### PR DESCRIPTION
**Pull request by KevinLiLu**
- Add `AtMinIsrPartitionCount` metric to `ReplicaManager`
- Add `AtMinIsr` metric to `Partition`
- Add `--at-min-isr-partitions` describe `TopicCommand` option

https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=103089398

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by KevinLiLu** [comment]
retest please

**Comment by KevinLiLu** [comment]
retest this please

**Comment by KevinLiLu** [comment]
Hi @harshach @vahidhashemian @lindong28 , can I get a review for this?

**Comment by KevinLiLu** [comment]
Bumping this as it has been a couple weeks. Can I get a review? @harshach @vahidhashemian @lindong28 @gwenshap

File: core/src/main/scala/kafka/admin/TopicCommand.scala [hunk 1]
```diff
@@ -636,11 +646,13 @@ object TopicCommand extends Logging {
       if(options.has(createOpt))
           CommandLineUtils.checkInvalidArgs(parser, options, replicaAssignmentOpt, Set(partitionsOpt, replicationFactorOpt))
       CommandLineUtils.checkInvalidArgs(parser, options, reportUnderReplicatedPartitionsOpt,
-        allTopicLevelOpts -- Set(describeOpt) + reportUnderMinIsrPartitionsOpt + reportUnavailablePartitionsOpt + topicsWithOverridesOpt)
+        allTopicLevelOpts -- Set(describeOpt) + reportUnderMinIsrPartitionsOpt + reportUnavailablePartitionsOpt + reportAtMinIsrPartitionsOpt + topicsWithOverridesOpt)
       CommandLineUtils.checkInvalidArgs(parser, options, reportUnderMinIsrPartitionsOpt,
-        allTopicLevelOpts -- Set(describeOpt) + reportUnderReplicatedPartitionsOpt + reportUnavailablePartitionsOpt + topicsWithOverridesOpt + zkConnectOpt)
+        allTopicLevelOpts -- Set(describeOpt) + reportUnderReplicatedPartitionsOpt + reportUnavailablePartitionsOpt + reportAtMinIsrPartitionsOpt + topicsWithOverridesOpt + zkConnectOpt)
```
Comment (gwenshap) [inline core/src/main/scala/kafka/admin/TopicCommand.scala]: Nit: 
We repeatedly add `reportUnderReplicatedPartitionsOpt`, `reportUnavailablePartitionsOpt`, `reportAtMinIsrPartitionsOpt` and `reportUnderMinIsrPartitionsOpt` to the set of disallowed options, and it is messy to read (and a  . Maybe create a new set like `allReplicationReportOpts` and use this everywhere?

**Review by gwenshap — CHANGES_REQUESTED** [review]
Looks good. One small request.

**Comment by KevinLiLu** [comment]
retest this please

File: core/src/main/scala/kafka/admin/TopicCommand.scala (hunk 1 above)
Comment (KevinLiLu) [reply core/src/main/scala/kafka/admin/TopicCommand.scala]: Great idea. Thanks!

**Comment by KevinLiLu** [comment]
Hi @gwenshap , thanks for the review. I have refactored the options code to use an `allReplicationReportOpts` Set.

**Comment by gwenshap** [comment]
Thank you. Looks good and tests pass. Merging.

**Pull request closed without merging**